js/bibox.js

```diff
@@ -422,6 +424,124 @@ module.exports = class bibox extends Exchange {
+        }
+        return this.parseTransactions (deposits, code, since, limit);
+    }
+
+    async fetchWithdrawals (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let symbol = undefined;
+        if (code !== undefined) {
+            const currency = this.currency (code);
+            symbol = currency['id'];
+        }
+        let size = (limit) ? limit : 100;
+        let response = await this.privatePostTransfer ({
+            'cmd': 'transfer/transferOutList',
+            'body': this.extend ({
+                'search': symbol,
+                'filter_type': 'ALL',   // ALL：all，1：in progress，2：error，3：completed
+                'page': 1,
+                'size': size,
+            }, params),
+        });
+        let withdrawals = this.safeValue (response['result'], 'items', []);
+        for (let i = 0; i < withdrawals.length; i++) {
+            withdrawals[i]['type'] = 'withdrawal';
+        }
+        return this.parseTransactions (withdrawals, code, since, limit);
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        //
+        // fetchDeposits
+        //      {'id': 1023291,
+        //      'coin_symbol': 'ETH',
+        //      'to_address': '0x72632f462ca5ea0aa28fa84167f0a61f87ac39e5',
+        //      'amount': '0.49170000',
+        //      'confirmCount': '16',
+        //      'createdAt': 1553123867000,
+        //      'status': 2}]                                                             }
+        //
+        // fetchWithdrawals
+        //
+        //       {      amount:  14,
+        //             address: "0x0123456789abcdef...",
+        //         successTime:  1514489710000,
+        //          addressTag: "",
+        //                txId: "0x0123456789abcdef...",
+        //                  id: "0123456789abcdef...",
+        //               asset: "ETH",
+        //           applyTime:  1514488724000,
+        //              status:  6                       }
+        //
+        let id = this.safeString (transaction, 'id');
+        let address = this.safeString (transaction, 'to_address');
+        let code = undefined;
+        let currencyId = this.safeString (transaction, 'coin_symbol');
+        if (currencyId in this.currencies_by_id) {
+            currency = this.currencies_by_id[currencyId];
+        } else {
+            code = this.commonCurrencyCode (currencyId);
+        }
+        if (currency !== undefined) {
+            code = currency['code'];
+        }
+        let timestamp = this.safeString (transaction, 'createdAt');
+        let type = this.safeString (transaction, 'type');
+        let status = this.parseTransactionStatusByType (this.safeString (transaction, 'status'), type);
+        let amount = this.safeFloat (transaction, 'amount');
+        return {
+            'info': transaction,
+            'id': id,
+            'txid': undefined,
```

Comment:
We need txid here.

Reply:
txid is not provided by bibox, so we should remove it?

        //       {'id': 521844,
        //       'coin_symbol': 'ETH',
        //       'to_address': '0xfd4e....',
        //       'addr_remark': '',
        //       'amount': '0.39452750',
        //       'fee': '0.00600000',
        //       'createdAt': 1553226906000,
        //       'memo': '',
        //       'status': 3}
        //

Reply:
Is that a withdrawal structure or a deposit structure? Can you paste a withdrawal structure here?

> so we should remove it?

Nope, we should keep it undefined if there's no `txid` in withdrawals.

Reply:
ok I have added it back thanks

Reply:
> Is that a withdrawal structure or a deposit structure? Can you paste a withdrawal structure here?
> 
> > so we should remove it?
> 
> Nope, we should keep it undefined if there's no `txid` in withdrawals.

structures as below if you haven't seen already.
        //
        // fetchDeposits
        //      {'id': 1023291,
        //      'coin_symbol': 'ETH',
        //      'to_address': '0x7263....',
        //      'amount': '0.49170000',
        //      'confirmCount': '16',
        //      'createdAt': 1553123867000,
        //      'status': 2}                                                             }
        //
        // fetchWithdrawals
        //
        //       {'id': 521844,
        //       'coin_symbol': 'ETH',
        //       'to_address': '0xfd4e....',
        //       'addr_remark': '',
        //       'amount': '0.39452750',
        //       'fee': '0.00600000',
        //       'createdAt': 1553226906000,
        //       'memo': '',
        //       'status': 3}
        //